pkg/kubectl/rollback.go

```diff
@@ -221,32 +224,22 @@ func (r *DaemonSetRollbacker) Rollback(obj runtime.Object, updatedAnnotations ma
 	if !ok {
 		return "", fmt.Errorf("passed object is not a DaemonSet: %#v", obj)
 	}
-	versionedExtensionsClient := versionedExtensionsClientV1beta1(r.c)
 	versionedAppsClient := versionedAppsClientV1beta1(r.c)
-	versionedDS, allHistory, err := controlledHistories(versionedExtensionsClient, versionedAppsClient, ds.Namespace, ds.Name)
+	versionedExtensionsClient := versionedExtensionsClientV1beta1(r.c)
+	versionedObj, allHistory, err := controlledHistories(versionedAppsClient, versionedExtensionsClient, ds.Namespace, ds.Name, "DaemonSet")
 	if err != nil {
 		return "", fmt.Errorf("unable to find history controlled by DaemonSet %s: %v", ds.Name, err)
 	}
+	versionedDS, ok := versionedObj.(*externalextensions.DaemonSet)
+	if !ok {
+		return "", fmt.Errorf("unable to perform type assertion for DaemonSet %s", ds.Name)
```

Comment:
If `!ok`, it means that the value doesn't hold the type. So the error message should say that the thing is not of DaemonSet type. Something like `"unexpected non-DaemonSet object returned: %v", versionedDS`. 

Same for other type assertion error messages.